### PR TITLE
Updated City and Favorites, added mock data and networking

### PR DESCRIPTION
**Pull request by CoraJacobson**
• Favorites feature - data model updated, FavoritesVC redesigned, segue from Favorites to CityDashboard implemented
• updated City model and added decoder
• added mock data loader, implemented networking and mock data for City

### CODE REVIEW

**Review by Kstu24 — APPROVED** [review]
Looks good! I like what you did with the mock data. I'm interested on learning how to do that as well.

**Comment by CoraJacobson** [comment]
[As an iOS app user, I want to pin cities to my profile, so that I can access the information without having to do another search.](https://trello.com/c/EuPRe8Le/10-as-an-ios-app-user-i-want-to-pin-cities-to-my-profile-so-that-i-can-access-the-information-without-having-to-do-another-search)

**Comment by CoraJacobson** [comment]
[As an iOS app user, I want to be able to search for a city, so that I can get information about it.](https://trello.com/c/VOgJ0AHH/7-as-an-ios-app-user-i-want-to-be-able-to-search-for-a-city-so-that-i-can-get-information-about-it)

File: labs-ios-starter/CityDashboardViewController.swift [hunk 1]
```diff
@@ -19,19 +19,48 @@ class CityDashboardViewController: UIViewController, UICollectionViewDataSource,
...
+    var favorites: [Favorite] = []
+    var favorited: Favorite? = nil
     var controller: ApiController?
-    
+    let context = (UIApplication.shared.delegate as! AppDelegate).persistentContainer.viewContext
+    // var cityProperties: [Int?] {
+    //     let crime = controller?.stringToInt(word: city?.crime ?? "")
+    //     let airQuality = controller?.stringToInt(word: city?.airQuality ?? "")
+    //     return [city?.livability, city?.walkability, city?.traffic, airQuality, crime, city?.rentalPrice]
+    // }
```
Comment (johnnydhicks) [inline labs-ios-starter/CityDashboardViewController.swift:32]: Remove

File: labs-ios-starter/Favorites/FavoritesViewController.swift [hunk 2]
```diff
@@ -55,29 +84,32 @@ extension FavoritesViewController: UICollectionViewDelegate, UICollectionViewDat
...
-
-        cell.walkabilityLabel.text = "Walkability: \(favorites![indexPath.row].walkabilityScore)"
-
+        cell.cityTitleLabel.text = favorites![indexPath.row].cityName! + ", " + favorites![indexPath.row].stateName!
         return cell
     }
 
     func collectionView(_ collectionView: UICollectionView, layout collectionViewLayout: UICollectionViewLayout, sizeForItemAt indexPath: IndexPath) -> CGSize {
-        return CGSize(width: view.frame.width, height: view.frame.height / 3)
+        return CGSize(width: view.frame.width, height: view.frame.height / 8)
```
Comment (johnnydhicks) [inline labs-ios-starter/Favorites/FavoritesViewController.swift:94]: Indentation looks off here, though could be from the view in GitHub.

File: labs-ios-starter/MapScreen/MapScreenViewController.swift [hunk 3]
```diff
@@ -11,191 +11,191 @@ import MapKit
...
+//            annotationView.glyphImage = UIImage(systemName: "house.fill")
+//        case "For Sale":
+//            annotationView.markerTintColor = UIColor(named: "LightBlue")
+//            annotationView.glyphImage = UIImage(systemName: "dollarsign.circle.fill")
+//        default:
+//            annotationView.markerTintColor = UIColor(named: "LightBlue")
+//        }
+//        return annotationView
+//    }
+//}
```
Comment (johnnydhicks) [inline labs-ios-starter/MapScreen/MapScreenViewController.swift:201]: If you aren't going to use this, just remove it it

File: labs-ios-starter/Model/Map.swift [hunk 4]
```diff
@@ -11,8 +11,8 @@ import UIKit
...
-struct Map {
-    var long: CLLocationDegrees = 0
-    var lat: CLLocationDegrees = 0
-    var cityName: String = ""
-}
+//struct Map {
+//    var long: CLLocationDegrees = 0
+//    var lat: CLLocationDegrees = 0
+//    var cityName: String = ""
+//}
```
Comment (johnnydhicks) [inline labs-ios-starter/Model/Map.swift:18]: Remove

**Pull request merged**